ee/server/service/software_installers.go

```diff
@@ -1668,38 +1677,102 @@ func (svc *Service) softwareBatchUpload(
 				ValidatedLabels:    p.ValidatedLabels,
 			}
 
-			// set the filename before adding metadata, as it is used as fallback
-			var filename string
-			cdh, ok := headers["Content-Disposition"]
-			if ok && len(cdh) > 0 {
-				_, params, err := mime.ParseMediaType(cdh[0])
-				if err == nil {
-					filename = params["filename"]
+			// check if we already have the installer based on the SHA256
+			teamIDs, err := svc.ds.GetSoftwareInstallerByHash(ctx, p.SHA256)
+			if err != nil {
+				return err
+			}
+
+			var tmID uint
+			if teamID != nil {
+				tmID = *teamID
+			}
+
+			foundInstaller, ok := teamIDs[tmID]
+
+			switch {
+			case ok:
+				installer.StorageID = p.SHA256
+				installer.Extension = foundInstaller.Extension
+				installer.Filename = foundInstaller.Filename
+				installer.Version = foundInstaller.Version
+				installer.Platform = foundInstaller.Platform
+				installer.Source = foundInstaller.Source
+				if foundInstaller.BundleIdentifier != nil {
+					installer.BundleIdentifier = *foundInstaller.BundleIdentifier
+				}
+				installer.Title = foundInstaller.Title
+			case !ok && len(teamIDs) > 0:
```

Comment:
We need to make sure that the associated user has access to the team the installer is on, as GitOps can now be called by team-specific users.

Also, if a URL is provided, we should limit matching installers to ones that explicitly include that same URL. Otherwie you can throw a URL into a YAML file for an installer that doesn't match the hash, and we'll behave as if we downloaded the installer and the hash matched, which isn't what we want.

Probably easiest to handle this at the data store layer while keeping existing logic.

Also worth a comment to explain why (setting the software title to whatever is already there) we have what looks like nearly duplicate code between the "team match" case and the "other team" case.